magicctapipe/io/tests/test_io_monly.py

```diff
@@ -17,456 +19,564 @@
+        assert "off_axis" in data.columns
+        assert "true_event_class" not in data.columns
+
+    def test_load_train_data_files_g(self, gamma_stereo_monly):
+        """
+        Check dictionary
+        """
+
+        events = load_train_data_files(str(gamma_stereo_monly[0]))
+        assert list(events.keys()) == ["M1_M2"]
+        data = events["M1_M2"]
+        assert np.all(data["combo_type"] == 0)
+        assert "off_axis" in data.columns
+        assert "true_event_class" not in data.columns
+
+    def test_load_train_data_files_off(self, gamma_stereo_monly):
+        """
+        Check off-axis cut
+        """
+        events = load_train_data_files(
+            str(gamma_stereo_monly[0]), offaxis_min="0.2 deg", offaxis_max="0.5 deg"
         )
-        assert np.all(data["gammaness"] > 0.1)
-
-
-def test_load_mc_dl2_data_file_opt(p_dl2_monly, gamma_dl2_monly):
-    """
-    Check on event_type
-    """
-    dl2_mc = [p for p in gamma_dl2_monly.glob("*")] + [p for p in p_dl2_monly.glob("*")]
-    for file in dl2_mc:
-
-        data_m, _, _ = load_mc_dl2_data_file(
-            str(file), "width>0", "magic_only", "simple"
+        data = events["M1_M2"]
+        assert np.all(data["off_axis"] >= 0.2)
+        assert np.all(data["off_axis"] <= 0.5)
+
+    def test_load_train_data_files_exc(self, temp_train_exc):
+        """
+        Check on exceptions
+        """
+        with pytest.raises(
+            FileNotFoundError,
+            match="Could not find any DL1-stereo data files in the input directory.",
+        ):
+            _ = load_train_data_files(str(temp_train_exc))
+
+    def test_load_train_data_files_tel_p(self, p_stereo_monly, config_gen):
+        """
+        Check dictionary
+        """
+
+        events = load_train_data_files_tel(str(p_stereo_monly[0]), config_gen)
+        assert list(events.keys()) == [2, 3]
+        data = events[2]
+        assert "off_axis" in data.columns
+        assert "true_event_class" not in data.columns
+
+    def test_load_train_data_files_tel_g(self, gamma_stereo_monly, config_gen):
+        """
+        Check dictionary
+        """
+
+        events = load_train_data_files_tel(str(gamma_stereo_monly[0]), config_gen)
+        assert list(events.keys()) == [2, 3]
+        data = events[3]
+        assert "off_axis" in data.columns
+        assert "true_event_class" not in data.columns
+
+    def test_load_train_data_files_tel_off(self, gamma_stereo_monly, config_gen):
+        """
+        Check off-axis cut
+        """
+        events = load_train_data_files_tel(
+            str(gamma_stereo_monly[0]),
+            config=config_gen,
+            offaxis_min="0.2 deg",
+            offaxis_max="0.5 deg",
         )
+        data = events[2]
+        assert np.all(data["off_axis"] >= 0.2)
+        assert np.all(data["off_axis"] <= 0.5)
 
-        assert np.all(data_m["combo_type"] == 3)
+    def test_load_train_data_files_tel_exc(self, temp_train_exc, config_gen):
```

Comment:
I think this test is also the same as in test_io.py - if the exception is raised due to missing files for 3tel analysis it should also work for MAGIC-only

Reply:
removed